core/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -519,7 +519,14 @@ public Settings archiveUnknownOrInvalidSettings(
             try {
                 Setting<?> setting = get(entry.getKey());
                 if (setting != null) {
-                    setting.get(settings);
+                    try {
+                        setting.get(settings);
+                    } catch (IllegalArgumentException ex) {
+                        if (setting.isMandatory()) {
+                            throw new IllegalStateException("can't archive mandatory setting [" + setting.getKey() + "]", ex);
```

Comment:
> setting [..] is invalid and mandatory

makes no sense to me, it's an illegal state because X and  X is that a mandatory setting can't be archived. it's caused by Y